### DISCUSSION TITLE
Welcome to SmartPendant Discussions!

### DISCUSSION THREAD

**Discussion by nickshl** [nickshl/SmartPendant#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Romsb** [comment]
Здравствуйте большое спасибо за классный проект!, а старые версии прошивки без скриптов где модно найти?

**Reply by Romsb** [reply]
хотя зачем, а скажите MB85RC256V FRAM нужно ставить?

**Reply by nickshl** [reply]
Я уверен что старые версии как-то можно достать из истории комитетов на GitHub.
Вместо FRAM MB85RC256V можно поставить EEPROM, например AT25C256C.

**Reply by Romsb** [reply]
конечно нужно, иначе ошибка NVRAM и настройки не сохраняются :)

**Reply by Romsb** [reply]
ещё раз большое спасибо за проекты!

**Comment by Romsb** [comment]
Ещё вопрос при включении картинка появляется, затем видно идёт перезапуск контроллера если нажать по экрану, а дальше всё вроде нормально работает. Это у меня так или так должно быть, прошивка последняя 0.030.0 25 MHz

**Reply by nickshl** [reply]
Если пульт подключен к контроллеру и нажата кнопка "MPG" то пульт считывает настройки с контроллера и перерисовывет всё. Это может выглядеть как "перезапуск". Если же просто нажатие в любое место экрана и происходит перезапуск то так быть не должно.

**Reply by Romsb** [reply]
![IMG_20251209_184912](https://github.com/user-attachments/assets/e6680565-4456-4c31-98b3-7186d560bd69)
Да подключен, всегда подключен, они распаяны на одной плате, плате жк экрана и распаяны 2 см витого провода. Всё так и есть жму "MPG" идёт перерисовка. Спасибо!

**Comment by Romsb** [comment]
А почему отказались от HM-11?

**Reply by Romsb** [reply]
глупый вопрос, не потянет он

**Reply by nickshl** [reply]
У HM-11 максимальная скорость заявлена 6KB(против 11.2KB у UART), плюс надо батарейку, плюс зарядное, плюс надёжность меньше чем у провода ну и если оси покрутить перед запуском ещё ничего если вдруг батарейка сядет, то вот запускать программу с SD карточки через Bluetooth было бы страшновато.

**Comment by Romsb** [comment]
Скажите, а как можно сделать, чтобы при включении пульт сразу подключался на управление к grblhal

**Reply by Romsb** [reply]
Наверное автор обиделся на меня, прошу прощения и ещё раз спасибо за хороший проект!

**Reply by nickshl** [reply]
Никто не обижался, просто сейчас празники. Сейчас так сделать нельзя, наверное надо будет настройку добавить.

**Comment by Romsb** [comment]
[SmartPendant.zip](https://github.com/user-attachments/files/24397748/SmartPendant.zip)
Вот если автор не против добавил настройку Auto MPG on start, вроде ничего не сломал
Автозапрос управления после старта: через задержку ~500 мс (чтобы UART/GrblComm успели подняться), затем вызывается grbl_comm.GainControl() и кнопка MPG становится “нажатой”
